src/veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.cc

```diff
@@ -0,0 +1,267 @@
+//
+// Documentation for these modules is at http://veins.car2x.org/
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#include <algorithm>
+
+#include "veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.h"
+
+#include "veins/modules/mobility/traci/TraCIConstants.h"
+
+namespace veins {
+namespace TraCISubscriptionManagement {
+
+PersonSubscriptionManager::PersonSubscriptionManager()
+    : SubscriptionManagerBase()
+    , mUpdatedPersons()
+    , mDisappearedPersons()
+{
+}
+
+bool PersonSubscriptionManager::updateWithList(std::list<std::string>& currentlyActivePersonIds) {
+    processPersonIDList(currentlyActivePersonIds);
+    return true;
+}
+
+bool PersonSubscriptionManager::update(TraCIBuffer& buffer) {
+
+    bool idListUpdateReceived = false;
+
+    // this is the object id that this subscription result contains
+    // content about. for example a person id.
+    std::string responseObjectID;
+    buffer >> responseObjectID;
+
+    // the number of response variables that are contained in this buffer
+    uint8_t numberOfResponseVariables;
+    buffer >> numberOfResponseVariables;
+    // this should either be one or five
+    ASSERT(numberOfResponseVariables == 1 || numberOfResponseVariables == 5);
+
+    // these need to be filled (total of 5 variables --> x, y count as one)
+    double x;
+    double y;
+    double speed;
+    std::string edge;
+    double angle;
+    std::string typeID;
+
+    for (int counter = 0; counter < numberOfResponseVariables; ++counter) {
+
+        // extract a couple of values:
+        // - identifies the variable (position, list etc.)
+        uint8_t responseVariableID;
+        buffer >> responseVariableID;
+        // - status of the variable
+        uint8_t variableStatus;
+        buffer >> variableStatus;
+        // - type of the variable
+        uint8_t variableType;
+        buffer >> variableType;
+
+        if (variableStatus == TraCIConstants::RTYPE_OK) {
+            // the status of the variable is ok
+
+            // now check which variable id we got
+            // it is either an id_list or a subscription for a specific vehicle
+            // never both
+            if (responseVariableID == TraCIConstants::ID_LIST) {
```

Comment:
This can be simplified a lot if you move handling `TraCIConstants::ID_LIST`s to it's own named method.

Reply:
That is done a few lines below.

Reply:
Alright.